### PR TITLE
Use default colours for examples

### PR DESCRIPTION
**Pull request by dstansby**
This took longer than I thought, but I think this is all the ugly examples changed to using colours from the default colour cycle. I also made a few small changes here and there if I spotted something, but nothing major.

### CODE REVIEW

File: examples/pyplots/fig_x.py [hunk 1]
```diff
@@ -1,20 +1,17 @@
 """
-=====
-Fig X
-=====
+========================
```
Comment (QuLogic) [inline examples/pyplots/fig_x.py]:
```suggestion
=======================
```
Doc builds broke ^.

File: examples/statistics/barchart_demo.py [hunk 2]
```diff
@@ -208,12 +208,13 @@ def plot_student_results(student, scores, cohort_size):
...
             'cohort_label': cohort_label}
 
+
 student = Student('Johnny Doe', 2, 'boy')
 scores = dict(zip(testNames,
                   (Score(v, p) for v, p in
                    zip(['7', '48', '12:52', '17', '14'],
                        np.round(np.random.uniform(0, 1,
-                                                  len(testNames))*100, 0)))))
+                                                  len(testNames)) * 100, 0)))))
```
Comment (anntzer) [inline examples/statistics/barchart_demo.py]: the linebreak here is a bit weird

**Comment by anntzer** [comment]
Looks like you committed an extra test2.png?  Unless I'm mistaken, it should be squashed away.

**Comment by jklymak** [comment]
This was a lot of work, but I'm not convinced changing all the examples from standard web color names to "tab:purple" etc is very user friendly.  I think I know how to use matplotlib, and even I looked at those color specifications and had to think about what the heck "tab:" meant for a couple of seconds; I can only imagine the naive user not understanding what is going on and why our colors need to be specified with a piece of magic in front of them.  Whats more, I can quickly look up on wikipedia what the default [web colors](https://en.wikipedia.org/wiki/Web_colors), I would need to spend a bit of time figuring out what the "tab:" colors are.  

I really apologize for this objection if this change was discussed and decided upon, but I am not convinced this is a positive change from a user's point of view.

**Comment by dstansby** [comment]
Don't worry, I kind of agree! In that case what I might do is put together a smaller PR where colours can be omitted altogether from the plotting, and then another PR that chooses "nice" colours (maybe black?) where they need to be explicitly specified.

**Comment by jklymak** [comment]
Well, maybe *some* of the examples stay with the "tab:x" colors (maybe with a comment explanation and link to the colors.py) and some get websafe colors and some get the short specifiers?  Overall, the idea of getting the colors to look better is a great idea.   I've not looked at the renders, but I imagine this *looks* a lot better.

File: examples/widgets/slider_demo.py [hunk 3]
```diff
@@ -46,15 +48,7 @@ def update(val):
 def reset(event):
     sfreq.reset()
     samp.reset()
-button.on_clicked(reset)
-
-rax = plt.axes([0.025, 0.5, 0.15, 0.15], facecolor=axcolor)
```
Comment (tacaswell) [inline examples/widgets/slider_demo.py]: Why remove the radio buttons?  Not _stricly_ in scope but not harmful.

**Comment by tacaswell** [comment]
Pushing to 3.2 as @dstansby marked it as needs revision.  If it is reading in time, re-milestone.  Moving out now to reduce the clutter in the 3.1 milestone.

**Comment by dstansby** [comment]
I'm going to close this since I don't have any plans to work on it in the future. I'll keep the branch alive though if anyone wants to take it up.

**Pull request closed without merging**